File: tests/rust/log_benchmark/src/points3d_large_batch.rs
use crate::points3d_shared::{Point3DInput, prepare_points3d};

const NUM_POINTS: usize = 50_000_000;

fn execute(rec: &mut rerun::RecordingStream, input: Point3DInput) -> anyhow::Result<()> {
    re_tracing::profile_function!();

    let Point3DInput {
        positions,
        colors,
        radii,
        label,
    } = input;

    rec.log(
        "large_batch",
        &rerun::Points3D::new(positions)
            .with_colors(colors)
            .with_radii(radii)
            .with_labels([label]),
    )?;
    Ok(())
}

/// Log a single large batch of points with positions, colors, radii and a splatted string.
pub fn run(rec: &mut rerun::RecordingStream) -> anyhow::Result<()> {
    re_tracing::profile_function!();
    let input = std::hint::black_box(prepare_points3d(42, NUM_POINTS));
    execute(rec, input)
}

File: tests/cpp/log_benchmark/points3d_large_batch.cpp
#include <utility>

#include "benchmarks.hpp"
#include "points3d_shared.hpp"
#include "profile_scope.hpp"

#include <rerun.hpp>

constexpr int64_t NUM_POINTS = 50000000;

static void execute(Point3DInput input) {
    PROFILE_FUNCTION();

    rerun::RecordingStream rec("rerun_example_benchmark_points3d_large_batch");

    rec.log(
        "large_batch",
        rerun::Points3D(input.positions)
            .with_colors(input.colors)
            .with_radii(input.radii)
            .with_labels({input.label})
    );
}

void run_points3d_large_batch() {
    PROFILE_FUNCTION();
    auto input = prepare_points3d(42, NUM_POINTS);
    execute(std::move(input));
}
